core/src/main/java/org/apache/druid/data/input/impl/DelimitedInputFormat.java

```diff
@@ -117,6 +122,18 @@ protected SeparateValueInputFormat(
     }
   }
 
+  private static Format getFormat(String delimiter)
+  {
+    if (",".equals(delimiter)) {
+      return Format.CSV;
+    } else if (delimiter != null && delimiter.length() > 0) {
```

Comment:
Since only single character delimiters are supported, suggest adding a check for this to the `DelimitedInputFormat` constructor that throws an exception if a multichar delimiter is provided, otherwise the behavior might be mysterious to users (specifying a delimiter but it's silently ignored)

Reply:
Solved.